modules/openapi-generator/src/main/resources/python/python-experimental/signing.mustache

```diff
@@ -0,0 +1,317 @@
+        """
+
+    def get_http_signature_headers(self, host, resource_path, method, headers, body, query_params):
+        """
+        Create a cryptographic message signature for the HTTP request and add the signed headers.
+        
+        :param resource_path : A string representation of the HTTP request resource path.
+        :param method: A string representation of the HTTP request method, e.g. GET, POST.
+        :param headers: A dict containing the HTTP request headers.
+        :param body: The string representation of the HTTP request body.
+        :param query_params: A string representing the HTTP request query parameters.
+        :return: A dict of HTTP headers that must be added to the outbound HTTP request.
+        """
+        if method is None:
+            raise Exception("HTTP method must be set")
+        if resource_path is None:
+            raise Exception("Resource path must be set")
+
+        signed_headers_dict, request_headers_dict = self.get_signed_header_info(
+            host, resource_path, method, headers, body, query_params)
+
+        header_items = [
+            "{0}: {1}".format(key.lower(), value) for key, value in signed_headers_dict.items()]
+        string_to_sign = "\n".join(header_items)
+
+        digest, digest_prefix = self.get_message_digest(string_to_sign.encode())
+        b64_signed_msg = self.sign_digest(digest)
+
+        request_headers_dict[HEADER_AUTHORIZATION] = self.get_authorization_header(
+            signed_headers_dict, b64_signed_msg)
+
+        return request_headers_dict
+
+    def load_private_key(self):
+        """Load the private key used to sign HTTP requests.
+            The private key is used to sign HTTP requests as defined in
+            https://datatracker.ietf.org/doc/draft-cavage-http-signatures/.
+        """
+        if self.private_key is not None:
+            return
+        with open(self.private_key_path, 'rb') as f:
+            # Decode PEM file and determine key type from PEM header.
+            # Supported values are "RSA PRIVATE KEY" and "ECDSA PRIVATE KEY".
+            keys = pem.parse(f.read())
+            if len(keys) != 1:
+                raise Exception("File must contain exactly one private key")
+            key = keys[0].as_text()
+            if key.startswith('-----BEGIN RSA PRIVATE KEY-----'):
+                self.private_key = RSA.importKey(key)
+            elif key.startswith('-----BEGIN EC PRIVATE KEY-----'):
+                self.private_key = ECC.importKey(key)
+            else:
+                raise Exception("Unsupported key")
+
+    def get_unix_time(self, ts):
+        """Converts and returns a datetime object to UNIX time, the number of seconds
+           elapsed since January 1, 1970 UTC.
+        """
+        return (ts - datetime.datetime(1970,1,1)).total_seconds())
+
+    def get_signed_header_info(self, host, resource_path, method, headers, body, query_params):
```

Comment:
Can change this method's name to add a `__` prefix so we can tell that it is not a publicly used method?
See http://python.robasworld.com/private-member-name-mangling/

Reply:
done